crates/drainer/src/utils.rs

```diff
@@ -99,11 +122,15 @@ pub fn parse_stream_entries<'a>(
         .into_report()
 }
 
-pub fn increment_stream_index(index: u8, total_streams: u8) -> u8 {
+pub fn increment_stream_index(index: u8, jobs_picked: u8, total_streams: u8) -> (u8, u8) {
```

Comment:
It would be better if you could either use a struct (since it is not clear what the 2 `u8`s represent), or document what each element in the tuple returned represents.